src/opencl_o5logon_fmt_plug.c

```diff
@@ -280,6 +281,10 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 			"failed in clEnqueueWriteBuffer buffer_keys");
 
 	BENCH_CLERROR(
+		clEnqueueWriteBuffer(queue[gpu_id], buffer_out, CL_FALSE, 0, sizeof(cl_uint), result, 0, NULL, multi_profilingEvent[1]),
+		"failed in clEnqueueWriteBuffer buffer_idx");
+
+	BENCH_CLERROR(
```

Comment:
I don't like these news at all. It probably means many other formats has potential problems. BTW I also tried putting a global memory barrier after L59 in the kernel and it didn't help! Given that, how am I supposed to depend on anything if thread n completes before thread 0 even starts?! The atomic functions are pretty much useless this way.